nomad/structs/structs.go

```diff
@@ -2849,6 +2849,14 @@ func (a *AllocatedResources) Comparable() *ComparableResources {
 	for _, r := range a.Tasks {
 		c.Flattened.Add(r)
 	}
+	// Add network resources that are at the task group level
+	if len(a.Shared.Networks) > 0 {
+		for _, network := range a.Shared.Networks {
```

Comment:
`range` would handle if the Networks is empty or nil I believe, so no need to explicitly check for length.